lib/document.js

```diff
@@ -496,6 +502,20 @@ class Document extends Readable {
     return this._write(object.toString() + '\n\n')
   }
 
+  _updateOutlinesCount(id) {
+    if (this._outlines[id].data.count < 1) {
+      this._outlines[id].data.count -= 1
+      this._outlines[id].prop('Count', this._outlines[id].data.count)
+    } else {
+      this._outlines[id].data.count = -1
+      this._outlines[id].prop('Count', this._outlines[id].data.count)
+    }
+    const nextParent = this._outlines[id].data.parentIndex
+    if(nextParent !== undefined) {
+      this._updateOutlinesCount(nextParent)
+    }
+  }
```

Comment:
I only skimmed through the PDF spec for Outlines and just wanted to check whether it is correct that the count written to the document is negative? E.g. from the test pdf:

```
2 0 obj
<<
	/Type /Outlines
	/First 4 0 R
	/Last 18 0 R
	/Count -15
>>
endobj
```

Reply:
From the spec:
_If the outline items is open, __Count__ is the sum of the number of visible descendent outline items at all levels. [...]  
If the outline item is closed, __Count__ is negative and its absolute value is the number of descendants that would be visible if the outline item were opened_

I  chose to set the tree closed by default, thus the count is negative. I may set the count of the root element to a positive value. Or all of them, if you prefer.

Reply:
Thanks for the clarification. Using the negative approach is perfectly fine 👍